### DISCUSSION TITLE
LoadImage from memory

### DISCUSSION THREAD

**Discussion by FJonske** [Project-MONAI/MONAI#7979] [Q&A]
Hello,

I am in the unusual situation of having more data than I can locally save, accessible via a modern and fast PACS implementation. As a consequence, I will have the DICOM instances of my 3D image series available directly in memory. I'd like to not write them to the disk and later read them via LoadImaged, but rather immediately make use of them, as that is much faster (with the amount of data I have, bottlenecking myself at the data loading stage may make a big difference). For the sake of argument, and since it's also not particularly slow, I could also convert the contents of those DICOM instances into something like a Nifti1Image.

Is there a clean way to skip saving and then using LoadImaged in my scenario? Or do I need to manually construct the MetaTensor or whatever LoadImaged would normally spit out? Any insights are appreciated.

**Comment by KumoLiu** [comment]
Hi @FJonske, you can directly use the data to do the later transform chains without LoadImage if your data is directly in the memory.
Thanks.

**Comment by FJonske** [comment]
I know I could do transforms on objects that aren't a MetaTensor, but some operations require the metadata (some operations may assume an identity for the affine matrix if you don't specify one), so I wanted to somehow get to the MetaTensor. I ended up subclassing LoadImage and modifying my version to just take instances instead of filenames, but otherwise leaving everything the same, so that I would still be given a MetaTensor at the end.
